platform/ios/src/MGLMapboxEvents.m

```diff
@@ -1,643 +1,161 @@
+    if (NSProcessInfo.processInfo.mgl_isInterfaceBuilderDesignablesAgent) {
+        return nil;
     }
-    return ([[NSUserDefaults standardUserDefaults] boolForKey:MGLMapboxMetricsEnabled] &&
-            [[NSUserDefaults standardUserDefaults] integerForKey:MGLMapboxAccountType] == 0 &&
-            !isLowPowerModeEnabled);
-#endif
-}
-
-
-- (BOOL)debugLoggingEnabled {
-    return (self.canEnableDebugLogging &&
-            [[NSUserDefaults standardUserDefaults] boolForKey:@"MGLMapboxMetricsDebugLoggingEnabled"]);
+    
+    static dispatch_once_t onceToken;
+    static MGLMapboxEvents *_sharedInstance;
+    dispatch_once(&onceToken, ^{
+        _sharedInstance = [[self alloc] init];
+    });
+    return _sharedInstance;
 }
 
-- (instancetype) init {
+- (instancetype)init {
     self = [super init];
     if (self) {
-        _currentAccountTypeValue = @0;
-        _currentMetricsEnabledValue = YES;
+        _eventsManager = [[MMEEventsManager alloc] init];
+        _eventsManager.debugLoggingEnabled = [[NSUserDefaults standardUserDefaults] boolForKey:MGLMapboxMetricsDebugLoggingEnabled];
+        _eventsManager.accountType = [[NSUserDefaults standardUserDefaults] integerForKey:MGLMapboxAccountType];
+        _eventsManager.metricsEnabled = [[NSUserDefaults standardUserDefaults] boolForKey:MGLMapboxMetricsEnabled];
         
-        _appBundleId = [[NSBundle mainBundle] bundleIdentifier];
-        _apiClient = [[MGLAPIClient alloc] init];
-
-        NSString *uniqueID = [[NSProcessInfo processInfo] globallyUniqueString];
-        _serialQueue = dispatch_queue_create([[NSString stringWithFormat:@"%@.%@.events.serial", _appBundleId, uniqueID] UTF8String], DISPATCH_QUEUE_SERIAL);
-
-        _locationManager = [[MGLLocationManager alloc] init];
-        _locationManager.delegate = self;
-        _paused = YES;
-        [self resumeMetricsCollection];
-
-        // Events Control
-        _eventQueue = [[NSMutableArray alloc] init];
-
-        // Setup Date Format
-        _rfc3339DateFormatter = [[NSDateFormatter alloc] init];
-        NSLocale *enUSPOSIXLocale = [[NSLocale alloc] initWithLocaleIdentifier:@"en_US_POSIX"];
-
-        [_rfc3339DateFormatter setLocale:enUSPOSIXLocale];
-        [_rfc3339DateFormatter setDateFormat:@"yyyy'-'MM'-'dd'T'HH':'mm':'ssZ"];
-        // Clear Any System TimeZone Cache
-        [NSTimeZone resetSystemTimeZone];
-        [_rfc3339DateFormatter setTimeZone:[NSTimeZone systemTimeZone]];
-
-        // Configure logging
-        if ([self isProbablyAppStoreBuild]) {
-            self.canEnableDebugLogging = NO;
-
-            if ([[NSUserDefaults standardUserDefaults] boolForKey:@"MGLMapboxMetricsDebugLoggingEnabled"]) {
-                NSLog(@"Telemetry logging is only enabled in non-app store builds.");
-            }
-        } else {
-            self.canEnableDebugLogging = YES;
+        // Because we can get here [MGLAccountManaber +load] it is possible that values from user defaults will be read and
```

Comment:
typo:

> [MGLAccountManaber +load]

Reply:
This `+` would typically go outside the brackets, as well.